jbehave-core/src/main/java/org/jbehave/core/embedder/StorySplitter.java

```diff
@@ -13,34 +14,37 @@
  * Splits story into the list of stories based on examples table configuration
  */
 public class StorySplitter {
-    public static List<Story> splitStories(List<Story> stories) {
+    public static List<Story> splitStories(List<Story> stories, String storyIndexFormat) {
         List<Story> splitStories = new ArrayList<Story>();
         for (Story story: stories) {
             if (story.getLifecycle().getExamplesTable().getRowCount() > 1) {
-                splitStories.addAll(splitStory(story));
+                splitStories.addAll(splitStory(story, storyIndexFormat));
             } else {
                 splitStories.add(story);
             }
         }
         return splitStories;
     }
 
-    private static List<Story> splitStory(Story story) {
+    private static List<Story> splitStory(Story story, String storyIndexFormat) {
         List<Story> splitStories = new ArrayList<Story>();
         List<Map<String, String>> rows = story.getLifecycle().getExamplesTable().getRows();
         for (int i = 0; i < rows.size(); i++) {
             ExamplesTable examplesTable = ExamplesTable.empty().withRows(Collections.singletonList(rows.get(i)));
             Lifecycle originLifecycle = story.getLifecycle();
             Lifecycle lifecycle = new Lifecycle(examplesTable, originLifecycle.getBefore(), originLifecycle.getAfter());
             String originPath = story.getPath();
-            Story splitStory = new Story(story, indexStory(originPath, i), lifecycle);
-            splitStory.namedAs(indexStory(story.getName(), i));
+
+            DecimalFormat decimalFormat = new DecimalFormat(storyIndexFormat);
+            String index = decimalFormat.format(i);
+            Story splitStory = new Story(story, indexStory(originPath, index), lifecycle);
+            splitStory.namedAs(indexStory(story.getName(), index));
             splitStories.add(splitStory);
         }
         return splitStories;
     }
 
-    private static String indexStory(String story, int index) {
-        return story.replace(".story", String.format(" [%d].story", index));
+    private static String indexStory(String story, String index) {
+        return story.replace(".story", String.format(" [%s].story", index));
```

Comment:
also I'm thinking of giving more freedom here: to allow users configure full part of the  index sub-string: "[%d]". In this case it would be possible for put some text or change overall format, e.g. "- %d" or "(index = %d)".

Reply:
> also I'm thinking of giving more freedom here: to allow users configure full part of the index sub-string: "[%d]". In this case it would be possible for put some text or change overall format, e.g. "- %d" or "(index = %d)".

But these cases can be implemented and in the current code as well.
For example "storyIndexFormat" with value "index=000" will give to us " StorySorting [index=000]", " StorySorting [index=001]", " StorySorting [index=002]" etc.

Reply:
but I can't change brackets: ` [...].story`

Reply:
> but I can't change brackets: ` [...].story`

Let me make sure that I understood correctly - we leave the .story extension (it is cut off later anyway)